action.yml

```diff
@@ -71,4 +75,4 @@ outputs:
 
 runs:
   using: docker
-  image: Dockerfile
+  image: Dockerfile.${{ inputs.lua-version }}
```

Comment:
Not sure if this work, depends if `image: Dockerfile` is some special trigger to start docker build instead of attempting docker pull.

Probably better to test before breaking tests of every project that is using this runner.

Reply:
Seems it isn't working as documented, here's even example that isn't far from this use case (and should be exactly same for expressions): https://docs.github.com/en/actions/creating-actions/creating-a-docker-container-action#creating-an-action-metadata-file

Failure is not with docker runner but with expression evaluation, either with `inputs` or `github`: 
```
Unrecognized named-value: 'inputs'. Located at position 1 within expression: inputs.lua-version
```
and
```
Unrecognized named-value: 'github'. Located at position 1 within expression: github.event.inputs.lua-version
```